### PR TITLE
Basic tox chat

### PR DESCRIPTION
**Pull request by suhr**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -176,10 +186,79 @@ class Input {
...
+
+        if(action === null) { return }
+
+        let response = null
+        switch (action.action) {
+            case "info":
+                response = this.tox.info()
+                break;
+            case "add":
+                if(action.message !== null) {
```
Comment (kpp) [inline src/index.ts:228]: No need to check it because you check in inside .addFriend

File: src/index.ts [hunk 2]
```diff
@@ -59,10 +55,27 @@ class Tox {
...
     addFriend(toxId: string, message?: string): Promise<ToxResponse> {
-        throw new Error("Unimplemented")
+        return new Promise((accept, reject) => {
+            let request: ToxRequest;
+
+            if(message !== undefined) {
+                request = { request: "AddFriend", tox_id: toxId, message }
+            }
+            else {
+                request = { request: "AddFriendNorequest", tox_id: toxId }
```
Comment (kpp) [inline src/index.ts:65]: How about an optional `message` field in AddFriend?

Comment (suhr) [reply src/index.ts:65]: I want to keep the protocol close to tox API.

File: src/index.ts (hunk 1 above)
Comment (suhr) [reply src/index.ts:228]: null ≠ undefined

Comment (kpp) [reply src/index.ts:228]: Then what's the point of `message?: string`?

File: src/index.ts (hunk 2 above)
Comment (kpp) [reply src/index.ts:65]: OK

File: src/index.ts (hunk 1 above)
Comment (suhr) [reply src/index.ts:228]: You can use `tox.addFriend(toxId)` instead of `tox.addFriend(toxId, null)`.

Comment (kpp) [reply src/index.ts:228]: > null ≠ undefined

What's the point then? Did you try?

Comment (suhr) [reply src/index.ts:228]: In the world of js there's no point. I could try using `undefined` instead of `null` in `Action` though.

**Pull request merged**